components/sup/src/manager/service/mod.rs

```diff
@@ -491,32 +338,20 @@ impl Service {
         }
     }
 
-    /// Run file_updated hook if present
-    pub fn file_updated(&self) -> bool {
-        if self.initialized {
-            match self.run_hook(HookType::FileUpdated) {
-                Ok(()) => {
-                    outputln!(preamble self.service_group, "File update hook succeeded");
-                    return true;
-                }
-                Err(err) => {
-                    outputln!(preamble self.service_group, "File update hook failed: {}", err);
-                }
-            }
-        }
-        false
-    }
-
     /// Run initialization hook if present
     pub fn initialize(&mut self) {
-        if !self.initialized {
-            outputln!(preamble self.service_group, "Initializing");
-            if let Some(err) = self.run_hook(HookType::Init).err() {
-                outputln!(preamble self.service_group, "Initialization failed: {}", err);
-                return;
-            }
-            self.initialized = true;
+        if self.initialized {
+            return;
+        }
+        outputln!(preamble self.service_group, "Initializing");
+        self.hooks.load_hooks(&self.supervisor.runtime_config,
+                              &self.config,
+                              &self.service_group);
+        if let Some(err) = self.run_hook(HookType::Init).err() {
```

Comment:
Great catch, I've got some changes on my branch to push up to optimize calling of compilation which will sort out these bugs from sneaking in for the future. Thanks for reviewing this for me ;)